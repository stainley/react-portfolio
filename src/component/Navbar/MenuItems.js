export const MenuItems = [
    {
        title: 'Home',
        url: '#',
        cName: 'nav-links'
    },
    {
        title: 'About Me',
        url: '/about',
        cName: 'nav-links'
    },
    {
        title: 'Portfolio',
        url: '/portfolio',
        cName: 'nav-links'
    },
    {
        title: 'Contact Me',
        url: '/contact',
        cName: 'nav-links'
    },
    {
        title: 'Sign Up',
        url: '#',
        cName: 'nav-links-mobile'
    },
];
