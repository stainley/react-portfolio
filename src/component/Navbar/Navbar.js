import React, {Component} from "react";
import {MenuItems} from './MenuItems';
import {Link} from 'react-router-dom';
import './Navbar.css';
import {Button} from "./Button";

export default class Navbar extends Component {

    state = {clicked: false}

    handleClick = () => {
        this.setState({clicked: !this.state.clicked})
    }

    render() {
        return (
            <nav data-testid="navbar" className="NavbarItems">
                <h1 className="navbar-logo">Stainley Lebron <i className="fab fa-react"/></h1>
                <div className="menu-icon" onClick={this.handleClick}>
                    <i className={this.state.clicked ? 'fas fa-times' : 'fas fa-bars'}/>
                </div>
                <ul className={this.state.clicked ? 'nav-menu active' : 'nav-menu'}>
                    {MenuItems.map((item, index) => {
                        return (
                            <li key={index}>
                                <Link to={item.url} className={item.cName}>
                                    {item.title}
                                </Link>
                            </li>
                        );
                    })}
                </ul>
                <Button>Sign Up</Button>
            </nav>
        );
    }
}
