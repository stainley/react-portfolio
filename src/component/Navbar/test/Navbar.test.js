import React from "react";
import ReactDom from 'react-dom';

import Navbar from "../Navbar";
import {render, cleanup} from "@testing-library/react";
import '@testing-library/jest-dom';

import renderer from 'react-test-renderer';

afterEach(cleanup);

describe('Testing Navbar', ()=> {
    it('renders without crashing', function () {
        const div = document.createElement("div");
        ReactDom.render(<Navbar />, div);
    });

    it('render navbar correctly', () => {
        const {getByTestId} = render(<Navbar />);

        expect(getByTestId('navbar')).toHaveTextContent('Stainley Lebron HomeAbout MePortfolioContact MeSign UpSign Up');
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<Navbar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
