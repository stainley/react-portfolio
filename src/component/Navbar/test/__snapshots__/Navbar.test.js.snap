// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar matches snapshot 1`] = `
<nav
  className="NavbarItems"
  data-testid="navbar"
>
  <h1
    className="navbar-logo"
  >
    Stainley Lebron 
    <i
      className="fab fa-react"
    />
  </h1>
  <div
    className="menu-icon"
    onClick={[Function]}
  >
    <i
      className="fas fa-bars"
    />
  </div>
  <ul
    className="nav-menu"
  >
    <li>
      <a
        className="nav-links"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="nav-links"
        href="/about"
        onClick={[Function]}
      >
        About Me
      </a>
    </li>
    <li>
      <a
        className="nav-links"
        href="/portfolio"
        onClick={[Function]}
      >
        Portfolio
      </a>
    </li>
    <li>
      <a
        className="nav-links"
        href="/contact"
        onClick={[Function]}
      >
        Contact Me
      </a>
    </li>
    <li>
      <a
        className="nav-links-mobile"
        href="/"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </li>
  </ul>
  <button
    className="btn btn-primary btn--medium"
  >
    Sign Up
  </button>
</nav>
`;
